fix: click on emoji closes edit or reply

aweiss-dev opened the pull request:
Clicking outside the text area while editing something or writing a reply will close the current draft and revert the input back to the default input.

Since the emoji picker was outside the text area, clicking on an emoji closed the current edit/response draft.

This got fixed by adding the emoji container to the "outside textarea click - ignore list".

While doing this, I recognized that the ignore list had several other components' class names hardcoded. This is highly dangerous since that functionality will break if the class names of the other components get changed. This lead me to do some small refactoring to class names for that specific cases.

In the future, we should avoid using hard-coded class names and instead rely on variables. This still is dangerous since our less files which hold the same identifiers need to have them hard coded, but with further migration to CSS modules or other CSS-in-JS solutions, this can be solved.

otto-the-bot committed:
chore: Update translations (#14105)

thisisamir98 committed:
runfix: Show correct count in typing indicator and add data-uie-names (#14158)

thisisamir98 committed:
feat: Adjust styles for typing indicator (ACC-309) (#14160)

* feat: Adjust styles for typing indicator (ACC-309)

* absolute position

thisisamir98 committed:
fix: Discard typing events after 1 minute (#14165)

* fix: Discard typing events after 1 minute

* change naming

atomrc committed:
runfix: Update window title live as unread conversations change (#14168)

V-Gira committed:
runfix: prevent overlapping added users (#14174)

tlebon committed:
runfix: use length for sidebar state instead of at for backcompat (#14177)

arjita-mitra committed:
chore: Bump application version

aweiss-dev committed:
fix: click on emoji closes edit or reply

aweiss-dev committed:
fix: click on emoji closes edit or reply

aweiss-dev committed:
chore: remove not needed strings

aweiss-dev committed:
chore: remove not needed strings

przemvs commented on src/script/components/InputBar/InputBar.tsx:
```diff
@@ -898,8 +901,8 @@ const InputBar = ({
-      id="conversation-input-bar"
-      className={cx('conversation-input-bar', {'is-right-panel-open': isRightSidebarOpen})}
+      id={conversationInputBarClassName}
+      className={cx(`${conversationInputBarClassName}`, {'is-right-panel-open': isRightSidebarOpen})}
```
I think it can be like that :)

```suggestion
      className={cx(conversationInputBarClassName, {'is-right-panel-open': isRightSidebarOpen})}
```

aweiss-dev replied:
👍

aweiss-dev committed:
chore: remove not needed strings

aweiss-dev committed:
chore: remove not needed strings

atomrc commented:
I think we should try the other way around and a a `data-*` for on the elements that need to block the outsideclick and avoid exporting classnames. 
What do you think?

V-Gira committed:
runfix: address design review of the font size slider [ACC-302] (#14266)

aweiss-dev committed:
feat: version without relying on classNames

aweiss-dev committed:
feat: version without relying on classNames

aweiss-dev commented:
> I think we should try the other way around and a a `data-*` for on the elements that need to block the outsideclick and avoid exporting classnames. What do you think?

I changed this now, and I like it better already. 
I am a bit sad that we don't use styled-components because it is easier to implement generic wrappers through the ```as prop``` handling it has included (https://emotion.sh/docs/styled#as-prop). 
In this case the wrapper is not generic and just implemented as a div.

atomrc commented on src/script/components/InputBar/util/clickHandlers.tsx:
```diff
@@ -0,0 +1,41 @@
+}
+const IgnoreClickWrapper = ({children, ...rest}: IgnoreClickWrapperProps) => {
+  return (
+    <div {...rest} data-ignore-click>
```
Let's make it obvious that it's related to the outside click handler.

```suggestion
    <div {...rest} data-outside-click-ignore>
```

atomrc commented on src/script/components/InputBar/util/clickHandlers.tsx:
```diff
@@ -0,0 +1,41 @@
+};
+
+const handleClickOutsideOfInputBar = (event: Event, callback: () => void): void => {
+  const ignoredParent = (event.target as HTMLElement).closest('div[data-ignore-click]') !== null;
```
```suggestion
  const ignoredParent = (event.target as HTMLElement).closest('div[data-outside-click-ignore]') !== null;
```

atomrc commented on src/script/components/InputBar/util/clickHandlers.tsx:
```diff
@@ -0,0 +1,41 @@
+  );
+};
+
+const handleClickOutsideOfInputBar = (event: Event, callback: () => void): void => {
```
```suggestion
const handleOutsideClick = (event: Event, callback: () => void): void => {
```

I don't think we need to mention the input bar at this point. This handler is more generic than this (and could potentially be used in many places)

atomrc commented on src/script/components/InputBar/util/clickHandlers.tsx:
```diff
@@ -0,0 +1,41 @@
+interface IgnoreClickWrapperProps extends HTMLProps<HTMLDivElement> {
+  children: ReactNode;
+}
+const IgnoreClickWrapper = ({children, ...rest}: IgnoreClickWrapperProps) => {
```
```suggestion
const IgnoreOutsideClickWrapper = ({children, ...rest}: IgnoreClickWrapperProps) => {
```

tlebon committed:
runfix: old images missing domain (FS-1001) (#14268)

* runfix: domain finally defined

* chore: add getter

V-Gira committed:
runfix: correct mention font-size and edit icon in replies (#14267)

atomrc committed:
fix: Avoid double file pasting trigger (#14269)

atomrc committed:
chore: Remove antiscroll (#14272)

PatrykBuniX committed:
fix: user not able to send multiple files [FS-1276] (#14271)

* fix: user not able to send multiple files

* feat: enalbe multiple file upload on img/asset button click

atomrc committed:
refactor: Cleanup file dropping/pasting in InputBar (#14270)

* refactoring: Cleanup file dropping/pasting in InputBar

* add tests

PatrykBuniX committed:
feat: api v3 support for typing endpoint [FS-1311] (#14274)

* feat: api v3 support for typing endpoint

* chore: bump core

aweiss-dev committed:
refactor: dataAttribute

aweiss-dev committed:
refactor: dataAttribute

arjita-mitra committed:
feat: Typing Indicator settings - Turn on or Off(ACC-307) (#14273)

* feat: Typing Indicator settings - Turn on or Off(ACC-307)

* feat: typing indicator is not visible when user is typing but typing mode is disabled(ACC-307)

* fix: revert deleted hook dependecy(ACC-307)

* refactor: remove extra if(ACC-307)

* fix: code review comments(ACC-307)

* fix: rename configuration variable(ACC-307)

atomrc committed:
refactor: Improve file tree structure (#14275)

V-Gira committed:
runfix: prevent call button to stay on screen in responsive view [ACC-324] (#14278)

* runfix: prevent call button to stay on screen in responsive view [ACC-324]

atomrc committed:
fix: Remove scrollbar artifacts (#14280)

atomrc committed:
chore: Remove deprecated styling for context menu (#14281)

PatrykBuniX committed:
refactor: file upload (#14276)

* refactor: move upload methods to conversation component

* refactor: remove hof

* test: fix tests

* refactor: add checkFileSharingPermission hof back

* test: fix test

* refactor: move hof to separate util

* refactor: apply CR suggestions

atomrc committed:
refactor: Make FadingScrollbar a react component (#14279)

aweiss-dev committed:
Update src/script/components/InputBar/util/clickHandlers.tsx

Co-authored-by: Thomas Belin <thomasbelin4@gmail.com>

aweiss-dev committed:
Update src/script/components/InputBar/util/clickHandlers.tsx

Co-authored-by: Thomas Belin <thomasbelin4@gmail.com>

dependabot[bot] committed:
chore(deps): Bump @wireapp/commons from 5.0.3 to 5.0.4 in /server (#14285)

Bumps [@wireapp/commons](https://github.com/wireapp/wire-web-packages) from 5.0.3 to 5.0.4.
- [Release notes](https://github.com/wireapp/wire-web-packages/releases)
- [Commits](https://github.com/wireapp/wire-web-packages/compare/@wireapp/commons@5.0.3...@wireapp/commons@5.0.4)

---
updated-dependencies:
- dependency-name: "@wireapp/commons"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump typescript from 4.9.3 to 4.9.4 in /server (#14284)

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 4.9.3 to 4.9.4.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v4.9.3...v4.9.4)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump postcss-import from 15.0.1 to 15.1.0 (#14287)

Bumps [postcss-import](https://github.com/postcss/postcss-import) from 15.0.1 to 15.1.0.
- [Release notes](https://github.com/postcss/postcss-import/releases)
- [Changelog](https://github.com/postcss/postcss-import/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss-import/compare/15.0.1...15.1.0)

---
updated-dependencies:
- dependency-name: postcss-import
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump webpack-cli from 5.0.0 to 5.0.1 (#14288)

Bumps [webpack-cli](https://github.com/webpack/webpack-cli) from 5.0.0 to 5.0.1.
- [Release notes](https://github.com/webpack/webpack-cli/releases)
- [Changelog](https://github.com/webpack/webpack-cli/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-cli/compare/webpack-cli@5.0.0...webpack-cli@5.0.1)

---
updated-dependencies:
- dependency-name: webpack-cli
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump prettier from 2.8.0 to 2.8.1 (#14290)

Bumps [prettier](https://github.com/prettier/prettier) from 2.8.0 to 2.8.1.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/2.8.0...2.8.1)

---
updated-dependencies:
- dependency-name: prettier
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @wireapp/copy-config from 2.0.3 to 2.0.4 (#14292)

Bumps [@wireapp/copy-config](https://github.com/wireapp/wire-web-packages) from 2.0.3 to 2.0.4.
- [Release notes](https://github.com/wireapp/wire-web-packages/releases)
- [Commits](https://github.com/wireapp/wire-web-packages/compare/@wireapp/copy-config@2.0.3...@wireapp/copy-config@2.0.4)

---
updated-dependencies:
- dependency-name: "@wireapp/copy-config"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump caniuse-lite from 1.0.30001436 to 1.0.30001439 (#14293)

Bumps [caniuse-lite](https://github.com/browserslist/caniuse-lite) from 1.0.30001436 to 1.0.30001439.
- [Release notes](https://github.com/browserslist/caniuse-lite/releases)
- [Commits](https://github.com/browserslist/caniuse-lite/compare/1.0.30001436...1.0.30001439)

---
updated-dependencies:
- dependency-name: caniuse-lite
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump zustand from 4.1.4 to 4.1.5 (#14289)

Bumps [zustand](https://github.com/pmndrs/zustand) from 4.1.4 to 4.1.5.
- [Release notes](https://github.com/pmndrs/zustand/releases)
- [Commits](https://github.com/pmndrs/zustand/compare/v4.1.4...v4.1.5)

---
updated-dependencies:
- dependency-name: zustand
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump react-router-dom from 6.4.4 to 6.4.5 (#14297)

Bumps [react-router-dom](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router-dom) from 6.4.4 to 6.4.5.
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Changelog](https://github.com/remix-run/react-router/blob/main/packages/react-router-dom/CHANGELOG.md)
- [Commits](https://github.com/remix-run/react-router/commits/react-router-dom@6.4.5/packages/react-router-dom)

---
updated-dependencies:
- dependency-name: react-router-dom
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump @wireapp/react-ui-kit from 9.2.2 to 9.2.3 (#14300)

Bumps [@wireapp/react-ui-kit](https://github.com/wireapp/wire-web-packages) from 9.2.2 to 9.2.3.
- [Release notes](https://github.com/wireapp/wire-web-packages/releases)
- [Commits](https://github.com/wireapp/wire-web-packages/compare/@wireapp/react-ui-kit@9.2.2...@wireapp/react-ui-kit@9.2.3)

---
updated-dependencies:
- dependency-name: "@wireapp/react-ui-kit"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @types/node from 18.11.10 to 18.11.13 (#14291)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 18.11.10 to 18.11.13.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @types/jest from 29.2.3 to 29.2.4 (#14294)

Bumps [@types/jest](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/jest) from 29.2.3 to 29.2.4.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/jest)

---
updated-dependencies:
- dependency-name: "@types/jest"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump typescript from 4.9.3 to 4.9.4 (#14296)

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 4.9.3 to 4.9.4.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v4.9.3...v4.9.4)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @typescript-eslint/eslint-plugin (#14295)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.45.0 to 5.46.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.46.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump react-router from 6.4.4 to 6.4.5 (#14301)

Bumps [react-router](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router) from 6.4.4 to 6.4.5.
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Changelog](https://github.com/remix-run/react-router/blob/main/packages/react-router/CHANGELOG.md)
- [Commits](https://github.com/remix-run/react-router/commits/react-router@6.4.5/packages/react-router)

---
updated-dependencies:
- dependency-name: react-router
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump cspell from 6.16.0 to 6.17.0 (#14299)

Bumps [cspell](https://github.com/streetsidesoftware/cspell) from 6.16.0 to 6.17.0.
- [Release notes](https://github.com/streetsidesoftware/cspell/releases)
- [Changelog](https://github.com/streetsidesoftware/cspell/blob/main/CHANGELOG.md)
- [Commits](https://github.com/streetsidesoftware/cspell/compare/v6.16.0...v6.17.0)

---
updated-dependencies:
- dependency-name: cspell
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump @typescript-eslint/parser from 5.45.0 to 5.46.0 (#14298)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.45.0 to 5.46.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.46.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

arjita-mitra committed:
refactor: rename the property WIRE_TYPING_MODE to WIRE_TYPING_INDICATOR_MODE  (#14282)

* feat: Typing Indicator settings - Turn on or Off(ACC-307)

* feat: typing indicator is not visible when user is typing but typing mode is disabled(ACC-307)

* fix: revert deleted hook dependecy(ACC-307)

* refactor: remove extra if(ACC-307)

* fix: code review comments(ACC-307)

* fix: rename configuration variable(ACC-307)

* refactor: rename the property WIRE_TYPING_MODE to WIRE_TYPING_INDICATOR_MODE(ACC-307)

atomrc committed:
fix: Do not store accessToken on login [FS-1309] (#14303)

PatrykBuniX committed:
refactor: use disposable ref hook (#14302)

* refactor: new implementation of disposable ref

* refactor: use new disposable ref for detecting long quotes

* refactor: rename original function and fix tests

* refactor: improve message rendere key

* refactor: get rid of useDisposableRef hook in MemberMessage component

* refactor: remove useDisposableRef hook from TextMessageRenderer component

* refactor: remove useDisposableRef hook

phoenixhdd committed:
feat: update headings (#14258)

* feat: update headings

* update headings

* update left sidebar heading

przemvs commented on src/script/ui/ContextMenu.tsx:
```diff
@@ -146,48 +154,60 @@ const ContextMenu: React.FC<ContextMenuProps> = ({entries, defaultIdentifier = '
+                type="button"
+                data-uie-name={entry.identifier || defaultIdentifier}
+                title={entry.title || entry.label}
+                {...(entry.isDisabled
+                  ? undefined
+                  : {
+                      onClick: event => {
+                        event.preventDefault();
+                        cleanUp();
+                        entry.click?.(event.nativeEvent);
+                      },
+                      onMouseEnter: () => {
+                        setSelected(undefined);
+                      },
+                    })}
```
Hmm..

Why not: 
`{...(entry.isDisabled && ({ props })}`

aweiss-dev replied:
I'm not too fond of this approach. Hiding whats going on behind the value of a prop has no benefit in my eyes, its more complex to read and find out whats going on.
Doing important stuff visible and spreading the ...rest we don't care about, on the other hand, is ok.

V-Gira committed:
runfix: correct missing paragraph tags [ACC-321] (#14283)

* runfix: correct missing paragraph tags [ACC-321]

* Apply suggestions from code review

Co-authored-by: Przemysław Jóźwik <przemyslaw.jozwik@wire.com>

thisisamir98 committed:
feat: Screen Reader announcement for Typing Indicator (ACC-304) (#14305)

phoenixhdd committed:
feat: Update list tags structure (#14304)

* feat: Update list tags structure

* remove unecsessary code

atomrc committed:
refactor: Use delegate event handlers for message clicks (#14306)

atomrc committed:
refactor: Improve quote long text detection (#14307)

atomrc committed:
runfix: Address displaying conversation details when there a no participants (#14309)

atomrc committed:
runfix: Handle middle click for markdown links (#14311)

phoenixhdd committed:
feat: Forms accessibility (#14310)

* feat: Forms accessibility

* fix typo

phoenixhdd committed:
feat: Update HTML tags in Modals, AppLock and DeviceID (#14313)

* feat: Update HTML tags in Modals, AppLock and DeviceID

* Cr fixes

PatrykBuniX committed:
runfix: conversation component infinite rerender (#14314)

PatrykBuniX committed:
Merge branch 'master' into dev

atomrc committed:
refactor: Avoid state in FadingScrollbar (#14316)

PatrykBuniX committed:
refactor: create drop file area component (#14317)

* refactor: create drop file area component

* test: DropFileArea component

* chore: fix typo

tlebon committed:
runfix: update config version for website failures (#14318)

arjita-mitra committed:
fix: Firefox chat area accessibility issues, stop infinite loop for chat messages using arrow keys(last-first and vice versa), focus on non content messages interactive elements(ACC-268) (#14319)

* fix: firefox chat area accessibility issues, stop infinite loop for chat messages, focus on non content messages interactive elements(ACC-268)

* fix: shift+tab from input bar/tab from info bar is focusing on the entire message list(Firefox+scrollable chats only)

aweiss-dev committed:
Remove ref from MentionSuggestions

aweiss-dev committed:
Merge branch fix/fs-1277_emoji of github.com:wireapp/wire-webapp into fix/fs-1277_emoji

aweiss-dev committed:
fix: merge errors
